docs/develop/app-structure.md

```diff
@@ -32,16 +40,92 @@ On disk it might look like this:
     └── world.go                     // world service code
 ```
 
-Encore is not opinionated about whether you use a monolith or multiple services. However, it does solve most of the traditional drawbacks that come with building microservices.
+Encore is not opinionated about whether you use a monolith or multiple services. However, it does solve most of the
+traditional drawbacks that come with building microservices.
+
+## Define service code with sub-packages
+
+Within a service, it's possible to have multiple sub-packages. This is a good way to define components, helper
+functions, or other code for your functions, should you wish to do that. You can create as many subpackages in any
+kind of nested structure to build your service in any way you want.
+
+To create sub-packages, you create sub-directories within a service package. Subpackages are itnernal code for services,
+they are not service packages. Only service packages can define API endpoints, and sub-packages within services cannot
+themselves define APIs.
+You can however define an API in a service package that calls a function within a sub-package.
+
+For example, rather than define the entire logic for an endpoint in that endpoint's function, you can call functions
```

Comment:
I added this example here since I felt this section was a bit confusing? I think the goal here was to explain that you can have more subpackages than just the service package, hopefully it's clear. Should we maybe also mention the possibility to have internal packages?